### PR TITLE
Enhance properties lexer / parser

### PR DESCRIPTION
**Pull request by taodongl**
test case is from: https://en.wikipedia.org/wiki/.properties

### CODE REVIEW

File: properties/PropertiesLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,26 @@
+lexer grammar PropertiesLexer;
+
+COMMENT
+    : [!#] (~[\r\n])*
```
Comment (KvanTTT) [inline properties/PropertiesLexer.g4:4]:
```suggestion
    : [!#] ~[\r\n]*
```

File: properties/PropertiesLexer.g4 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+SLASH
+    : '\\'  -> more, pushMode(INSIDE)
+    ;
+CHARACTER
+    : ~ [!#:=\r\n]
+    ;
+
+mode INSIDE;
+SLASH_DELIMITER: ~[\r\n] -> type(CHARACTER),popMode;
+SLASH_JOINT: ('\r')? '\n' -> type(CHARACTER),popMode;
```
Comment (KvanTTT) [inline properties/PropertiesLexer.g4:21]:
```suggestion
SLASH_JOINT: '\r'? '\n' -> type(CHARACTER), popMode;
```

File: properties/PropertiesParser.g4 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    ;
+row
+    : comment
+    | line
+    ;
+line
+    : key DELIMITER value=key eol
+    | key DELIMITER eol
+    | key eol
+    ;
```
Comment (KvanTTT) [inline properties/PropertiesParser.g4:16]:
```suggestion
line
    : key (DELIMITER value=key?)? eol
    ;
```

File: properties/PropertiesLexer.g4 [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    ;
+SLASH
+    : '\\'  -> more, pushMode(INSIDE)
+    ;
+CHARACTER
+    : ~ [!#:=\r\n]
+    ;
+
+mode INSIDE;
+SLASH_DELIMITER: ~[\r\n] -> type(CHARACTER),popMode;
```
Comment (KvanTTT) [inline properties/PropertiesLexer.g4:20]:
```suggestion
mode INSIDE;

SLASH_DELIMITER: ~[\r\n] -> type(CHARACTER), popMode;
```

File: properties/PropertiesLexer.g4 [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+SLASH_JOINT: ('\r')? '\n' -> type(CHARACTER),popMode;
+
+mode VALUE_MODE;
+VALUE_TERM: ('\r')? '\n' -> type(NEWLINE),popMode;
```
Comment (KvanTTT) [inline properties/PropertiesLexer.g4:24]:
```suggestion
VALUE_TERM: '\r'? '\n' -> type(NEWLINE), popMode;
```

**Review by KvanTTT — COMMENTED** [review]
Welcome to grammars repository and thanks for the contribuiton! But it contains some minor formatting and simplification issues.

**Comment by lkishalmi** [comment]
@taodongl @KvanTTT What would be the license of this properties grammars?

**Comment by KvanTTT** [comment]
I don't know since I'm not the author of the grammar.

**Comment by taodongl** [comment]
@lkishalmi  No License, whatever you use

**Comment by KvanTTT** [comment]
> @lkishalmi No License, There are no restrictions, you can use whatever you want

Actually "No License" doesn't mean "no restrictions", see https://choosealicense.com/no-permission/ It's better always to add a license (I suggest adding **MIT**).

**Pull request merged**